opensrp-chw/src/main/java/org/smartregister/chw/interactor/DefaultPncHomeVisitInteractorFlv.java

```diff
@@ -677,6 +677,10 @@ public String getPreProcessed() {
                 pnc_visit_date.put("hint",
                         MessageFormat.format(pnc_visit_date.getString(JsonFormConstants.HINT), visitRule.getVisitName())
                 );
+
+                JSONObject pnc_hf_next_visit_date = JsonFormUtils.getFieldJSONObject(fields, "pnc_hf_next_visit_date");
+                pnc_hf_next_visit_date.put(JsonFormConstants.VALUE, DateTimeFormat.forPattern("dd-MM-yyyy").print(visitRule.getDueDate()));
```

Comment:
This is a bug, the visit due date you are using is for the current visit, not the next visit

Reply:
The `visitRule` from which the due date is retrieved is for the next visit and not the current one. 
In line 148 `PNCHealthFacilityVisitRule visitRule = PNCVisitUtil.getNextPNCHealthFacilityVisit(summary.getDeliveryDate(), summary.getLastVisitDate())`